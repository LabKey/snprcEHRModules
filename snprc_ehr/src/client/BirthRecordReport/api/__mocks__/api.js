/* eslint-disable no-unused-vars, camelcase */

import { NewAnimalData } from '../../tests/fixtures/apiTestData'

export const request = ({ schemaName, queryName, viewName = '', sort = '', columns = [], filterArray = [] }) => {
    // console.log(`loading list: ${queryName}`);
    return new Promise(resolve => {
        switch (queryName) {
        case 'NewAnimalData':
            resolve(NewAnimalData)
            break
        default:
            break
        }
    })
}
