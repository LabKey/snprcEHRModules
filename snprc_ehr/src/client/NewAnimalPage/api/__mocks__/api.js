/* eslint-disable no-unused-vars, camelcase */

import { CurrentSpeciesLookup, AccountLookup, validInstitutions, ValidDiet, AcquisitionTypesLookup,
    valid_bd_status, PotentialDams, PotentialSires, ActiveLocationsAll, colonyGroups,
    ProtocolLookup, pedigreeGroups } from '../../tests/fixtures/apiTestData'

export const request = ({ schemaName, queryName, viewName = '', sort = '', columns = [], filterArray = [] }) => {
    // console.log(`loading list: ${queryName}`);
    return new Promise(resolve => {
        switch (queryName) {
        case 'CurrentSpeciesLookup':
            resolve(CurrentSpeciesLookup)
            break
        case 'AccountLookup':
            resolve(AccountLookup)
            break
        case 'validInstitutions':
            resolve(validInstitutions)
            break
        case 'ValidDiet':
            resolve(ValidDiet)
            break
        case 'AcquisitionTypesLookup':
            resolve(AcquisitionTypesLookup)
            break
        case 'PotentialDams':
            resolve(PotentialDams)
            break
        case 'PotentialSires':
            resolve(PotentialSires)
            break
        case 'ActiveLocationsAll':
            resolve(ActiveLocationsAll)
            break
        case 'colonyGroups':
            resolve(colonyGroups)
            break
        case 'ProtocolLookup':
            resolve(ProtocolLookup)
            break
        case 'pedigreeGroups':
            resolve(pedigreeGroups)
            break
        case 'valid_bd_status':
            resolve(valid_bd_status)
            break
        default:
            break
        }
    })
}
