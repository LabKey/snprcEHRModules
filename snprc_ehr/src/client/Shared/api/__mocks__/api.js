/* eslint-disable no-unused-vars, camelcase */

import { NewAnimalData } from '../../../BirthRecordReport/tests/fixtures/apiTestData'
import {
    CurrentSpeciesLookup, AccountLookup, validInstitutions, ValidDiet, AcquisitionTypesLookup,
    valid_bd_status, PotentialDams, PotentialSires, ActiveLocationsAll, colonyGroups,
    ProtocolLookup, pedigreeGroups
} from '../../../NewAnimalPage/tests/fixtures/apiTestData'

export const request = ({ schemaName, queryName, viewName = '', sort = '', columns = [], filterArray = [] }) => {
    // console.log(`loading list: ${queryName}`);
    return new Promise(resolve => {
        switch (queryName) {
            case 'CurrentSpeciesLookup':
                resolve(CurrentSpeciesLookup)
                break
            case 'AccountLookup':
                resolve(AccountLookup)
                break
            case 'validInstitutions':
                resolve(validInstitutions)
                break
            case 'ValidDiet':
                resolve(ValidDiet)
                break
            case 'AcquisitionTypesLookup':
                resolve(AcquisitionTypesLookup)
                break
            case 'PotentialDams':
                resolve(PotentialDams)
                break
            case 'PotentialSires':
                resolve(PotentialSires)
                break
            case 'colonyGroups':
                resolve(colonyGroups)
                break
            case 'ProtocolLookup':
                resolve(ProtocolLookup)
                break
            case 'pedigreeGroups':
                resolve(pedigreeGroups)
                break
            case 'valid_bd_status':
                resolve(valid_bd_status)
                break
            case 'NewAnimalData':
                resolve(NewAnimalData)
                break
            default:
                break
        }
    })
}

export const executeSql = ({ schemaName, sql, sort = '' }) => {
    return new Promise(resolve => {
        if (sql.indexOf('FROM ehr_lookups.rooms') > 0) {
            resolve(ActiveLocationsAll)
        }
    })
}
