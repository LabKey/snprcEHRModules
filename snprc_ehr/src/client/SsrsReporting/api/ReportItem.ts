export interface ReportItem {
    id: number;
    label: string;
    value: string;
    description: string;
    parameters: string;
    rsParameters?:string;
}