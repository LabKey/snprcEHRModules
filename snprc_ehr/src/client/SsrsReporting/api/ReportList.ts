import { OptionsType } from 'react-select'

export interface ReportType {
    id: number;
    label: string;
    value: string;
    description: string;
    parameters: string;

}
export const reportList: OptionsType<ReportType> =  
    [
        {id: 1, label: 'Long Term Clinic Admissions', value: 'ReportS/report/LabkeyReports/LongTermClinic', description: 'Clinical admissions over 21 days',  parameters:'' },
        {id: 2, label: 'Overdue TB and Physical', value: 'ReportS/report/CenterReports/Routine%20Health/OverduePhyTB', description: 'Last TB test and physical exams that are over 180 days',  parameters:'' },
        {id: 3, label: 'Baboon weaning', value: 'ReportS/report/ColonyReports/Boon_weaning', description: 'Baboon infants that are ready to be weaned',  parameters:'' },
        {id: 4, label: 'Rhesus SPF weaning', value: 'ReportS/report/ColonyReports/rh2_weaning', description: 'Rhesus infants that are ready to be weaned', parameters:'' },
        {id: 5, label: 'Colony Census (current)', value: 'testReport1', description: 'Colony census as of today', parameters:'' },
        {id: 6, label: 'Colony Census (select date)', value: 'testReport2', description: 'Colony census on a specific date',  parameters:'targetDate:Date:Target Date' },
        {id: 7, label: 'Birth record report', value: 'ReportS/report/Development/BirthRecord', description: 'Birth certificate for animal',  parameters:'TargetID:string:Animal Id' },
        {id: 8, label: 'IACUC 3yr Rollover (Standard)', value: 'ReportS/report/CenterReports/IACUC/3YrRollover', description: 'Target Date is used to determine 3yr period NULL uses today.', parameters:'IACUC:string:IACUC, TargetDate:Date:Target Date, Mode:String:Mode:Standard' },
        {id: 8, label: 'IACUC 3yr Rollover (Backdate)', value: 'ReportS/report/CenterReports/IACUC/3YrRollover' ,description: 'Target Date sets the 3yr end date.  Start date is 3 years before target date.', parameters:'IACUC:string:IACUC, TargetDate:Date:Target Date, Mode:String:Mode:BackDate' },
        {id: 10, label: 'IACUC 3yr Rollover (Harddate)', value: 'ReportS/report/CenterReports/IACUC/3YrRollover', description: 'Target date sets the end date.  Start date is the date file start date before the target date. Could result in more or less than 3 yrs', parameters:'IACUC:string:IACUC, TargetDate:Date:Target Date, Mode:String:Mode:HardDate' }
    ];
    