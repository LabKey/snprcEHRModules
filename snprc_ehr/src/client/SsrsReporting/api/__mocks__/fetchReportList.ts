import { OptionsType } from 'react-select'
import { ReportItem } from '../ReportItem'

// API test data
//Report server docs: https://docs.microsoft.com/en-us/sql/reporting-services/url-access-parameter-reference?view=sql-server-ver15
export const fetchReportList = () => {  
    return new Promise<OptionsType<ReportItem>> ( (resolve, reject) => {
        resolve(
            [
                {id: 1, label: 'Long Term Clinic Admissions', value: 'LabkeyReports/LongTermClinic', description: 'Clinical admissions over 21 days',  parameters:'', rsParameters:'&rs:Command=Render' },
                {id: 2, label: 'Overdue TB and Physical', value: 'CenterReports/Routine%20Health/OverduePhyTB', description: 'Last TB test and physical exams that are over 180 days',  parameters:'', rsParameters: '&rs:Command=Render' },
                {id: 3, label: 'Baboon weaning', value: 'ColonyReports/Boon_weaning', description: 'Baboon infants that are ready to be weaned',  parameters:'', rsParameters: '&rs:Command=Render' },
                {id: 4, label: 'Rhesus SPF weaning', value: 'ColonyReports/rh2_weaning', description: 'Rhesus infants that are ready to be weaned', parameters:'', rsParameters: '&rs:Command=Render' },
                {id: 5, label: 'Colony Census (current)', value: 'testReport1', description: 'Colony census as of today', parameters:'', rsParameters: '&rs:Command=Render' },
                {id: 6, label: 'Colony Census (select date)', value: 'testReport2', description: 'Colony census on a specific date',  parameters:'targetDate:Date:Target Date', rsParameters: '&rs:Command=Render' },
                {id: 7, label: 'Birth record report', value: 'Development/BirthRecord', description: 'Birth certificate for animal',  parameters:'TargetID:string:Animal Id', rsParameters: '&rc:Parameters=Collapsed' },
                {id: 8, label: 'IACUC 3yr Rollover (Standard)', value: 'CenterReports/IACUC/3YrRollover&Mode=1', description: 'Target Date is used to determine 3yr period.', parameters:'IACUC:string:IACUC, TargetDate:Date:Target Date', rsParameters: '&rc:Parameters=Collapsed' },
                {id: 8, label: 'IACUC 3yr Rollover (Backdate)', value: 'CenterReports/IACUC/3YrRollover&Mode=2' ,description: 'Target Date sets the end date.  Start date is 3 years before target date.', parameters:'IACUC:string:IACUC, TargetDate:Date:Target Date', rsParameters: '&rc:Parameters=Collapsed' },
                {id: 10, label: 'IACUC 3yr Rollover (Harddate)', value: 'CenterReports/IACUC/3YrRollover&Mode=3', description: 'Target date sets the end date.  Start date is the start date before the target date. Could result in more or less than 3 yrs.', parameters:'IACUC:string:IACUC, TargetDate:Date:Target Date', rsParameters: '&rc:Parameters=Collapsed' }
            ]
        ), (error: string ) => {
            reject(error)
        }
    
    })
}
    