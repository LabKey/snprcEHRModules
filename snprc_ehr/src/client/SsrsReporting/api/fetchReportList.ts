import { ReportItem } from './ReportItem'
import { request } from '../../Shared/api/api'

const parse = rows => {
    return rows.map( ( { data } ) => {
        const reportItem: ReportItem = 
         { 
            id: data.Id.value, 
            sortOrder: data.SortOrder.value,
            label: data.Label.value,
            value: data.Report.value,
            description: data.Description.value,
            parameters: data.Parameters.value,
            rsParameters: data.rsParameters.value
        }
        return reportItem;
    })
}

export const fetchReportList = (): Promise<ReportItem[]> => {
    return new Promise((resolve, reject) => {
        request({
            schemaName: 'snprc_ehr',
            queryName: 'ExternalReports',
            columns: ['Id', 'SortOrder', 'Label', 'Report', 'Description', 'Parameters', 'rsParameters'],
            sort: 'SortOrder'
        }).then(({ rows }) => {
            const parsedRows:ReportItem[] = parse(rows)
            resolve(parsedRows)
        }).catch(error => {
            console.log('error', error)
            reject(error)
        })
    })
}

    