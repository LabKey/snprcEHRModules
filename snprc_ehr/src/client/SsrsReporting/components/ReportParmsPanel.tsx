import React from "react";
import { ParmItem } from "./ParmItem";
import { ReportParm } from "../services/parmUtils";
import { ReportItem } from "../api/ReportItem"

interface Props {
  reportParameters: ReportParm[];
  handleParmChange(index: number, parmItem: ReportParm): void;
  selectedReport: ReportItem;
}
export default class ReportParmsPanel extends React.Component<Props, {}> {

  handleParmChange = (index:number, parmItem: ReportParm): void => {
    this.props.handleParmChange(index, parmItem);
  };

  render() {
    return (
      <>
        <div className="wizard-panel-col">
        {this.props.selectedReport &&
          <div className="info-text-span"> {this.props.selectedReport.description} </div>
        }
          {this.props.reportParameters &&
           this.props.reportParameters.map(
              (parm, index) => {
                return <ParmItem 
                  parm={parm} 
                  index={index}
                  key={this.props.selectedReport.value+parm.name} 
                  handleParmChange = {this.handleParmChange}
                  />;
              }
           )}
          {this.props.reportParameters &&
            this.props.reportParameters.length === 0 && (
              <div style={{ padding: "1rem" }}>
                Report doesn't require parameters
              </div>
            )}
        </div>
      </>
    );
  }
}
