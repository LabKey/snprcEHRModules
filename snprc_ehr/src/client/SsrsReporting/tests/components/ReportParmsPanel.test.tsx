/* eslint-disable no-unused-vars */

import React from 'react'
import { shallow } from 'enzyme'
import moment from 'moment'
import ReportParmsPanel from '../../components/ReportSelectionPanel'
import { fetchReportList } from '../../api/fetchReportList'
import { OptionsType } from 'react-select'
import { ReportItem } from '../../api/ReportItem'

let wrapper

beforeEach( async () => {
    
    let reportList: OptionsType<ReportItem> =[]
    await fetchReportList().then ( (response) => {
        ( reportList = response)
    })
    .catch(err => {
        console.log(err)
    })

    wrapper = shallow(<ReportParmsPanel
        reportList={reportList}
        handleChange={(): void => {}}
        selectedReport={reportList[0]}
    />)
})

afterEach(() => {
    wrapper.unmount()
})

describe('SelectionParmsPanel tests', () => {
    test('Should render the ReportParmsPanel with selected Report.', () => {
        expect(wrapper).toMatchSnapshot()
    })

})
