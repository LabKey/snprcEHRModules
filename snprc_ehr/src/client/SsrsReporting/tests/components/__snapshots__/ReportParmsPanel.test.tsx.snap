// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectionParmsPanel tests Should render the ReportParmsPanel with selected Report. 1`] = `
<Fragment>
  <label
    className="field-label-align-close"
  >
    Select Report
  </label>
  <StateManager
    autoFocus={true}
    className="shared-dropdown"
    classNamePrefix="shared-select"
    defaultInputValue=""
    defaultMenuIsOpen={false}
    defaultValue={null}
    id="report-select"
    isLoading={false}
    onChange={[Function]}
    options={
      [
        {
          "description": "Clinical admissions over 21 days",
          "id": 1,
          "label": "Long Term Clinic Admissions",
          "parameters": "",
          "rsParameters": "&rs:Command=Render",
          "value": "LabkeyReports/LongTermClinic",
        },
        {
          "description": "Last TB test and physical exams that are over 180 days",
          "id": 2,
          "label": "Overdue TB and Physical",
          "parameters": "",
          "rsParameters": "&rs:Command=Render",
          "value": "CenterReports/Routine%20Health/OverduePhyTB",
        },
        {
          "description": "Baboon infants that are ready to be weaned",
          "id": 3,
          "label": "Baboon weaning",
          "parameters": "",
          "rsParameters": "&rs:Command=Render",
          "value": "ColonyReports/Boon_weaning",
        },
        {
          "description": "Rhesus infants that are ready to be weaned",
          "id": 4,
          "label": "Rhesus SPF weaning",
          "parameters": "",
          "rsParameters": "&rs:Command=Render",
          "value": "ColonyReports/rh2_weaning",
        },
        {
          "description": "Colony census as of today",
          "id": 5,
          "label": "Colony Census (current)",
          "parameters": "",
          "rsParameters": "&rs:Command=Render",
          "value": "testReport1",
        },
        {
          "description": "Colony census on a specific date",
          "id": 6,
          "label": "Colony Census (select date)",
          "parameters": "targetDate:Date:Target Date",
          "rsParameters": "&rs:Command=Render",
          "value": "testReport2",
        },
        {
          "description": "Birth certificate for animal",
          "id": 7,
          "label": "Birth record report",
          "parameters": "TargetID:string:Animal Id",
          "rsParameters": "&rc:Parameters=Collapsed",
          "value": "Development/BirthRecord",
        },
        {
          "description": "Target Date is used to determine 3yr period.",
          "id": 8,
          "label": "IACUC 3yr Rollover (Standard)",
          "parameters": "IACUC:string:IACUC, TargetDate:Date:Target Date",
          "rsParameters": "&rc:Parameters=Collapsed",
          "value": "CenterReports/IACUC/3YrRollover&Mode=1",
        },
        {
          "description": "Target Date sets the end date.  Start date is 3 years before target date.",
          "id": 8,
          "label": "IACUC 3yr Rollover (Backdate)",
          "parameters": "IACUC:string:IACUC, TargetDate:Date:Target Date",
          "rsParameters": "&rc:Parameters=Collapsed",
          "value": "CenterReports/IACUC/3YrRollover&Mode=2",
        },
        {
          "description": "Target date sets the end date.  Start date is the start date before the target date. Could result in more or less than 3 yrs.",
          "id": 10,
          "label": "IACUC 3yr Rollover (Harddate)",
          "parameters": "IACUC:string:IACUC, TargetDate:Date:Target Date",
          "rsParameters": "&rc:Parameters=Collapsed",
          "value": "CenterReports/IACUC/3YrRollover&Mode=3",
        },
      ]
    }
    placeholder="Select Report"
    value={
      {
        "description": "Clinical admissions over 21 days",
        "id": 1,
        "label": "Long Term Clinic Admissions",
        "parameters": "",
        "rsParameters": "&rs:Command=Render",
        "value": "LabkeyReports/LongTermClinic",
      }
    }
  />
</Fragment>
`;
